src/go/cmd/metadata-server/metadata.go

```diff
@@ -181,12 +189,85 @@ func (th *TokenHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Collect some extra data for diagnostics (aka which services rely on ADCs).
+	// User-Agent: "Fluent-Bit", "gcloud-golang/0.1"
+	ua := r.Header.Get("User-Agent")
+	pod := th.getPodNameByIP(r.Context(), ipPort[0])
```

Comment:
QQ: Assuming this gets really slow because we have many pods. How will it affect our system? Would it mean that some svcs will take longer to start up because it takes longer to get a token?

Reply:
The http server uses go routines. So this will not as such cause a lot of extra load, but will incur some extra latency if a service asks for ADCs right away. I don't have a better idea for this yet.

Reply:
Yeah, I was not concerned about system load, more about the startup latency. Most of our services create ADC-based clients on startup and do it synchronously.

I think ~1 second latency should be ok, esp. given it should not stack for the overall startup time.

+1 for the flag.